### ISSUE TITLE
Unable to initialize ckan postgresql db

### ISSUE THREAD

**Issue by oguya** [oguya/ansible-ckan#2]
Using `paster` with `plugin` option doesn't work in playbook & on shell:

```
- name: Initialize ckan postgresql database
  sudo_user: "{{ckan_user.name}}"
  command: "{{ ckan_virtualenv }}/bin/paster --plugin=ckan db init -c {{ ckan_conf_dir }}/production.ini"
```

It produces the following error:

```
TASK: [ckan | Initialize ckan postgresql database] **************************** 
failed: [ckan01] => {"changed": true, "cmd": ["/usr/lib/ckan/ckan.localhost.localdomain/bin/paster", "--plugin=ckan", "db", "init", "-c", "/etc/ckan/ckan.localhost.localdomain/production.ini"], "delta": "0:00:02.041566", "end": "2015-07-14 14:39:22.889204", "rc": 1, "start": "2015-07-14 14:39:20.847638", "warnings": []}
stderr: Traceback (most recent call last):
  File "/usr/lib/ckan/ckan.localhost.localdomain/bin/paster", line 9, in <module>
    load_entry_point('PasteScript==1.7.5', 'console_scripts', 'paster')()
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 104, in run
    invoke(command, command_name, options, args[1:])
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 143, in invoke
    exit_code = runner.run(args)
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 238, in run
    result = self.command()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/cli.py", line 208, in command
    self._load_config(cmd!='upgrade')
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/cli.py", line 148, in _load_config
    load_environment(conf.global_conf, conf.local_conf)
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/config/environment.py", line 232, in load_environment
    p.load_all(config)
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 124, in load_all
    unload_all()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 181, in unload_all
    unload(*reversed(_PLUGINS))
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 209, in unload
    plugins_update()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 116, in plugins_update
    environment.update_config()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/config/environment.py", line 270, in update_config
    search.check_solr_schema_version()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/search/__init__.py", line 306, in check_solr_schema_version
    % (version, ', '.join(SUPPORTED_SCHEMA_VERSIONS)))
ckan.lib.search.common.SearchError: SOLR schema version not supported: 1.5. Supported versions are [2.3]

FATAL: all hosts have already failed -- aborting
```

**Comment by oguya** [comment]
Running the command manually also produces a similar error:

```
(ckan.localhost.localdomain)ckan@ckan01:~/ckan.localhost.localdomain/src/ckan$ paster db init -c /etc/ckan/ckan.localhost.localdomain/production.ini 
Traceback (most recent call last):
  File "/usr/lib/ckan/ckan.localhost.localdomain/bin/paster", line 9, in <module>
    load_entry_point('PasteScript==1.7.5', 'console_scripts', 'paster')()
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 104, in run
    invoke(command, command_name, options, args[1:])
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 143, in invoke
    exit_code = runner.run(args)
  File "/usr/lib/ckan/ckan.localhost.localdomain/local/lib/python2.7/site-packages/paste/script/command.py", line 238, in run
    result = self.command()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/cli.py", line 208, in command
    self._load_config(cmd!='upgrade')
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/cli.py", line 148, in _load_config
    load_environment(conf.global_conf, conf.local_conf)
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/config/environment.py", line 232, in load_environment
    p.load_all(config)
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 124, in load_all
    unload_all()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 181, in unload_all
    unload(*reversed(_PLUGINS))
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 209, in unload
    plugins_update()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/plugins/core.py", line 116, in plugins_update
    environment.update_config()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/config/environment.py", line 270, in update_config
    search.check_solr_schema_version()
  File "/usr/lib/ckan/ckan.localhost.localdomain/src/ckan/ckan/lib/search/__init__.py", line 306, in check_solr_schema_version
    % (version, ', '.join(SUPPORTED_SCHEMA_VERSIONS)))
ckan.lib.search.common.SearchError: SOLR schema version not supported: 1.5. Supported versions are [2.3]
(ckan.localhost.localdomain)ckan@ckan01:~/ckan.localhost.localdomain/src/ckan$ 
```

**Comment by oguya** [comment]
Configuring ckan's solr core first & then initializing the database seems to fix this error

**Steps to reproduce**
- copy ckan's solr schema.xml to solr core conf dir & restart solr service
  
  ```
  - name: Configure solr core used by ckan
    command: /bin/cp -a {{ ckan_virtualenv }}/src/ckan/ckan/config/solr/schema.xml {{ solr_var_dir }}/data/{{ item }}/conf/schema.xml
    notify: reload solr
    with_items: solr_cores
    tags: solr
  ```
- initialize ckan postgresql db:
  
  ```
  (ckan.localhost.localdomain)ckan@ckan01:~/ckan.localhost.localdomain/src/ckan$ paster db init -c /etc/ckan/ckan.localhost.localdomain/production.ini 
  Initialising DB: SUCCESS
  ```

**Comment by oguya** [comment]
closed via 8b5b09d
